Update to rustc master

akiss77 opened the pull request:
* enum_set moved to collections,
* from_str deprecated.

alexcrichton commented:
Could this actually drop support for `enum_set` entirely? The libcollections crate is not slated to become stable just yet for 1.0, so we shouldn't be introducing a dependency on it.

akiss77 commented:
Since the PR consisted of two independent changes, I've split it into two and moved the drop of `enum_set` to https://github.com/rust-lang/rustc-serialize/pull/15 . This here is left to handle the deprecation of `std::str::from_str`.

alexcrichton commented:
Thanks again!
